### ISSUE TITLE
[HOLD for payment 2024-10-14] [$250] The intro text on private notes list stays fixed when it should scroll with the page

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#49298]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.36-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @dukenv0307 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1726513353005399

## Action Performed:
1. Open 1:1 chat
2. Click on the header
3. Add private notes with many lines
4. In private notes list page, scroll the page
## Expected Result:
The intro text should be attached to the content of the page and scroll like we did with many pages
## Actual Result:
The intro text is fixed when you scroll the page
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/d8e88e27-cf62-4883-bda0-3570f48a9494

https://github.com/user-attachments/assets/ccb4a18b-04b6-45f6-8474-ea26a588504c

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021835811668345209683</li>
        <li>Upwork Job ID: 1835811668345209683</li>
        <li>Last Price Increase: 2024-09-30</li>
<li>Automatic offers: </li>
<ul>
<li>brunovjk | Reviewer | 104226815</li>
<li>nkdengineer | Contributor | 104226816</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @lschurr</details>

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Gives error saying "Please enter a valid website using lowercase characters"

### What is the root cause of that problem?

We are not adding this `Text` [here](https://github.com/Expensify/App/blob/1156e9780fd9fb1de48c4913ea8db5a31a0c6e9e/src/pages/PrivateNotes/PrivateNotesListPage.tsx#L98-L104) in `ScrollView`


https://github.com/Expensify/App/blob/1156e9780fd9fb1de48c4913ea8db5a31a0c6e9e/src/pages/PrivateNotes/PrivateNotesListPage.tsx#L98-L104

### What changes do you think we should make in order to solve the problem?


We can create a function like `getHeaderMessage` for this Text and use it inside `ScrollView` or simpler we can move this `Text` to nside `ScrollView`

```
        <ScrollView
            contentContainerStyle={styles.flexGrow1}
            bounces={false}
        >
            <Text style={[styles.mb5, styles.ph5]}>{translate('privateNotes.personalNoteMessage')}</Text>
            {privateNotes.map((item) => getMenuItem(item))}
        </ScrollView>
```

### What alternative solutions did you explore? (Optional)
NA

**Comment by vr-varad** [comment]
Hey @brunovjk I would like to work on this issue.

**Comment by vr-varad** [comment]
Contributor details
Your Expensify account email: varadgupta21@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01148273080994adc1

**Comment by vr-varad** [comment]
I want to work on this issue. @brunovjk @lschurr

**Comment by markom01** [comment]
@m-natarajan This is not happening on macOS Safari as you stated in bug platforms:

https://github.com/user-attachments/assets/7e3f4063-8e48-4ee2-a961-1ba043bc9493

**Comment by markom01** [comment]
Contributor details
Your Expensify account email: [marko.meseldzija@hyperether.com](mailto:marko.meseldzija@hyperether.com)
Upwork Profile Link: https://www.upwork.com/freelancers/~011d144bd16d9790fc

**Comment by markom01** [comment]
Contributor details
Your Expensify account email: marko.meseldzija@hyperether.com
Upwork Profile Link: https://www.upwork.com/freelancers/~011d144bd16d9790fc

**Comment by bikramiter** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The intro text should be attached to the content of the page and scroll like we did with many pages
### What is the root cause of that problem?
The personalNoteMessage text is outside of the ScrollView as 
https://github.com/Expensify/App/blob/14b99ca0a12e9686818bc3e937f091199de69750/src/pages/PrivateNotes/PrivateNotesListPage.tsx#L98C13-L104C26
### What changes do you think we should make in order to solve the problem?
We can move the personalNoteMessage text inside the ScrollView as below.
```
<ScrollView
    contentContainerStyle={styles.flexGrow1}
    bounces={false}
>
    <Text style={[styles.mb5, styles.ph5]}>{translate('privateNotes.personalNoteMessage')}</Text>
    {privateNotes.map((item) => getMenuItem(item))}
</ScrollView>
```
### What alternative solutions did you explore? (Optional)
The current design is aligned with pages like Profile description page and Room description page. So, I am not sure if we should alter the behavior for this page. Profile description page also limits text to 1000 characters if we should consider this for this case.
<img width="1702" alt="image" src="https://github.com/user-attachments/assets/434b168a-2cf6-4b59-b991-47844f705a03">

**Comment by bikramiter** [comment]
Performed a minor edit to the problem statement.

**Comment by brunovjk** [comment]
> I want to work on this issue. @brunovjk @lschurr if it's open for contribution.

@vr-varad If you haven't already, please post a proposal and I will review then all shortly.

**Comment by vr-varad** [comment]
## Proposal
In the current code implementation:
```
<Text style={[styles.mb5, styles.ph5]}>{translate('privateNotes.personalNoteMessage')}</Text>
<ScrollView 
    contentContainerStyle={styles.flexGrow1} 
    bounces={false} 
> 
    {privateNotes.map((item) => getMenuItem(item))} 
</ScrollView>

```
The `<Text>` component (displaying the introductory message) is placed outside the <ScrollView>. This causes the text to remain fixed while only the ScrollView's content (i.e., the list of private notes) scrolls.

### Proposed Solution
To fix this issue, the introductory text should be moved inside the <ScrollView>. This way, the text will scroll along with the list of private notes.

The updated code should look like this:
```
<ScrollView
    contentContainerStyle={styles.flexGrow1}
    bounces={false}
>
    <Text style={[styles.mb5, styles.ph5]}>{translate('privateNotes.personalNoteMessage')}</Text>
    {privateNotes.map((item) => getMenuItem(item))}
</ScrollView>
```
Explanation of Changes:
1. Current Position: The `<Text>` component (introductory message) is positioned outside the `<ScrollView>`. This makes it fixed, meaning it doesn't scroll with the rest of the content.
2. Proposed Position: The `<Text>` component is moved inside the `<ScrollView>`. This ensures that both the message and the list of private notes are part of the same scrollable area.

**Comment by FitseTLT** [comment]
I think this is an expected behavior as the intro text is an informative text about the notes that is not part of the private notes list.

**Comment by brunovjk** [comment]
Thank you all for the proposals and investigation. I am having problems with my internet, the company is coming to replace my modem today. If this issue is urgent, please reassign it. Thank you very much.

**Comment by brunovjk** [comment]
@nkdengineer's [proposal](https://github.com/Expensify/App/issues/49298#issuecomment-2353845750) to move the introductory text inside the `<ScrollView>`, which would allow it to scroll along with the private notes content. This would "fix" the issue where the text currently stays fixed while scrolling. This behavior (the text being fixed) was introduced [here](https://github.com/brunovjk/Expensify-App/commit/dae4c62c0ffbe41b2fa7295818b9c0732fe7d30e#diff-0d05483fb9e164de27769acdee5548dcad876428e9996a35105e9794489eda6cR120).

Before proceeding with the fix, we would like to confirm:
- **Is the expected behavior for the intro text to scroll with the page, or should it remain fixed?**

Once we have confirmation, we can move forward with either applying the fix or leaving the current behavior unchanged. Thank you!

🎀👀🎀 C+ reviewed

**Comment by MonilBhavsar** [comment]
> Is the expected behavior for the intro text to scroll with the page, or should it remain fixed?

Agree, looks like feature and not a bug
The text area form field's description should not scroll with the text area. cc @Expensify/design for thoughts please.

**Comment by dubielzyk-expensify** [comment]
I usually think we only should have the app header stay fixed so in this instance I think the intro text should scroll. Keen to hear if @dannymcclain agrees though

**Comment by dannymcclain** [comment]
I do agree 👍

**Comment by brunovjk** [comment]
Thank you all. @MonilBhavsar I think we can go with @nkdengineer's https://github.com/Expensify/App/issues/49298#issuecomment-2353845750 then, thank you :D

**Comment by MonilBhavsar** [comment]
Thanks for the thoughts Design team! 🙇 
---

@brunovjk yes, I think we should update similar other pages(workspace and room description and may be others...) while we're here as they all would require same fix

**Comment by brunovjk** [comment]
Hey MonilBhavsar, I agree with you. It seems like none of the proposals fully address the issue you raised. I'm could search through the app to find the full list of places that need to be updated, or maybe we could ask the contributors to update their proposals with this info. What do you think?

**Comment by MonilBhavsar** [comment]
Agree.

@nkdengineer @vr-varad @bikramiter @markom01 if you all want to update the proposal based on this discussion https://github.com/Expensify/App/issues/49298#issuecomment-2372221756

**Comment by bikramiter** [comment]
Thanks @MonilBhavsar and @brunovjk. I further looked into the similar scenarios like Room description and Workspace profile description. 
- The edit behavior for all cases (Private notes, Room description and Workspace profile description) are same.
- After 'Save', for Room description and Workspace profile description, the intro text scrolls with the comment. Only for Private notes, the intro text does not scroll.

Please refer the screen recording explaining this for a large screen and ios screen. I could not attach this due to larger file size. https://drive.google.com/file/d/1_vTxA5MtDCn8w2Nc6395AHfieY4mQ4BU/view?usp=sharing

So, I feel making the below update([same as proposed earlier](https://github.com/Expensify/App/issues/49298#issuecomment-2355212458)) will handle the case. This will be aligned with the discussed behavior as well as consistant with the other pages.

```
<ScrollView
    contentContainerStyle={styles.flexGrow1}
    bounces={false}
>
    <Text style={[styles.mb5, styles.ph5]}>{translate('privateNotes.personalNoteMessage')}</Text>
    {privateNotes.map((item) => getMenuItem(item))}
</ScrollView>
```

**Comment by nkdengineer** [comment]
@MonilBhavsar I double checked other pages like Room description, Workspace profile description, ... and the intro text scrolls with the comment, the intro text doesn't scroll in only Private notes
And I think we just need to change in private notes page as in the [proposal](https://github.com/Expensify/App/issues/49298#issuecomment-2353845750) I mentioned

**Comment by brunovjk** [comment]
Not overdue. We asked on [Slack](https://expensify.slack.com/archives/C02NK2DQWUX/p1727674574871459) for opinions on how to proceed here. We are still discussing it :D

**Comment by MonilBhavsar** [comment]
Thanks all! Assigning @nkdengineer as they were the first to put up a proposal and proposal remains unchanged

**Comment by brunovjk** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
 
 * [x]  [@brunovjk] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/31344
 * [x]  [@brunovjk] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/31344#issuecomment-2410960081
 * [x]  [@brunovjk] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
 * [x]  [@brunovjk] Determine if we should create a regression test for this bug. **Yes**
 * [x]  [@brunovjk] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

### Regression Test Proposal

- Open 1:1 chat
- Click on the header
- Add private notes with many lines
- In private notes list page, scroll the page
- Verify that: The intro text should be attached to the content of the page and scroll like

Do we agree 👍 or 👎?

**Comment by brunovjk** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
 
 * [x]  [@brunovjk] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/31344
 * [x]  [@brunovjk] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/31344#issuecomment-2410960081
 * [x]  [@brunovjk] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
 * [x]  [@brunovjk] Determine if we should create a regression test for this bug. **Yes**
 * [x]  [@brunovjk] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

### Regression Test Proposal

- Open 1:1 chat
- Click on the header
- Add private notes with many lines
- In private notes list page, scroll the page
- Verify that: The intro text should be attached to the content of the page and scroll like

Do we agree 👍 or 👎?

cc: @lschurr

**Comment by lschurr** [comment]
Payment summary:

- Contributor: $250 @nkdengineer (paid via [Upwork](https://www.upwork.com/nx/wm/workroom/38606136/overview))
- Reviewer: $250 @brunovjk (paid via [Upwork](https://www.upwork.com/nx/wm/workroom/38582733/overview))
